plugins/destination/mysql/client/types.go

```diff
@@ -2,63 +2,84 @@ package client
 
 import (
 	"fmt"
+	"sort"
+	"strings"
 
-	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/cloudquery/plugin-sdk/v2/types"
 	"golang.org/x/exp/maps"
 )
 
-func SQLType(t schema.ValueType) string {
-	switch t {
-	case schema.TypeBool:
-		return "bool"
-	case schema.TypeInt:
-		return "bigint"
-	case schema.TypeFloat:
-		return "float"
-	case schema.TypeUUID:
-		return "binary(16)"
-	case schema.TypeByteArray:
-		return "blob"
-	case schema.TypeTimestamp:
-		return "datetime"
-	case schema.TypeCIDR,
-		schema.TypeMacAddr,
-		schema.TypeInet:
-		return "nvarchar(255)"
-	case schema.TypeString,
-		schema.TypeStringArray,
-		schema.TypeUUIDArray,
-		schema.TypeCIDRArray,
-		schema.TypeMacAddrArray,
-		schema.TypeInetArray,
-		schema.TypeIntArray:
-		return "text"
-	case schema.TypeJSON:
-		return "json"
-	default:
-		panic("unknown type " + t.String())
+func mySQLTypeToArrowType(tableName string, columnName string, sqlType string) (arrow.DataType, error) {
+	if strings.HasPrefix(sqlType, "datetime") {
+		// MySQL permits up to microseconds (6 digits) precision
+		return arrow.FixedWidthTypes.Timestamp_us, nil
 	}
-}
-
-func SchemaType(tableName string, columnName string, sqlType string) (schema.ValueType, error) {
-	sqlTypeToSchemaType := map[string]schema.ValueType{
-		"bool":          schema.TypeBool,
-		"tinyint(1)":    schema.TypeBool,
-		"bigint":        schema.TypeInt,
-		"bigint(20)":    schema.TypeInt,
-		"float":         schema.TypeFloat,
-		"binary(16)":    schema.TypeUUID,
-		"blob":          schema.TypeByteArray,
-		"datetime":      schema.TypeTimestamp,
-		"nvarchar(255)": schema.TypeInet,
-		"varchar(255)":  schema.TypeInet,
-		"text":          schema.TypeString,
-		"json":          schema.TypeJSON,
+	sqlTypeToSchemaType := map[string]arrow.DataType{
+		"bool":              arrow.FixedWidthTypes.Boolean,
+		"tinyint(1)":        arrow.FixedWidthTypes.Boolean,
```

Comment:
why this one is needed?

Reply:
This is more a defensive measure I think in case someone is doing MySQL -> MySQL sync and defined `bool` as `tinyint(1)` (similar to https://github.com/cloudquery/cloudquery/pull/9143/files). I think we can drop it for now

Reply:
OK so the correct reason was that we return `bool` a bit below so when we normalized we need to support `bool` too.
https://github.com/cloudquery/cloudquery/pull/10169/commits/ef4fdd18e07a15a2cb98cd3d77b12b9c2fa23fb5 should fix it.

See normalization code here https://github.com/cloudquery/cloudquery/pull/10169/files#diff-7a94fa4d0f7e4c8c80b1ad959ace684a43b44408b3d1d9cee3ea9ef2d87dd9a2R29